src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersTests.cs

```diff
@@ -0,0 +1,1171 @@
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg == null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (null == arg)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+    }
+
+    string this[string name]
+    {
+        get
+        {
+            if (name is null)
+                throw new ArgumentNullException(nameof(name));
+            return name;
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task ArgumentNullExceptionThrowIfNull_FixAppliesAsAppropriate()
+        {
+            await new VerifyCS.Test()
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestCode =
+@"
+using System;
+
+namespace System
+{
+    public class ArgumentNullException : Exception
+    {
+        public ArgumentNullException() { }
+        public ArgumentNullException(string paramName) { }
+        public ArgumentNullException(string paramName, string message) { }
+        public ArgumentNullException(string paramName, Exception innerException) { }
+        public static void ThrowIfNull(object argument, string name = null) { }           
+    }
+}
+
+class C
+{
+    void M(string arg)
+    {
+        {|CA1510:if (arg is null)
+            throw new ArgumentNullException(nameof(arg));|}
+        {|CA1510:if (arg == null)
+            throw new ArgumentNullException(nameof(arg), (string)null);|}
+        {|CA1510:if (null == arg)
+            throw new ArgumentNullException(nameof(arg));|}
+        {|CA1510:if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+        {|CA1510:if (arg == null)
+        {
+            throw new ArgumentNullException(nameof(arg), """");
+        }|}
+        {|CA1510:if (null == arg)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+        {|CA1510:if (null == arg)
+        {
+            throw new ArgumentNullException(""something else"");
+        }|}
+
+        if (arg == ""test"")
+        {
+            Console.WriteLine(arg);
+        }
+        else {|CA1510:if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }|}
+
+        if (arg is null)
+            throw new ArgumentNullException(nameof(arg), ""possibly meaningful message"");
+
+        if (arg is null)
+            throw new ArgumentNullException(nameof(arg), new Exception());
+
+        if (arg is null)
+        {
+            Console.WriteLine(); // another operation in the block
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg is not null) // inverted condition
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+
+        if (arg != null) // inverted condition
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
```

Comment:
NIT: we might want to cover the exception thrown in else part
```cs
    if (arg != null)
    {
        DoSomething();
    }
    else
    {
        throw new ArgumentNullException(nameof(arg));
    }
```
Fix:
```cs
    ArgumentNullException.ThrowIfNull(arg);
    DoSomething();
```

Reply:
We could.  I was just going for the 95% cases.  This is sufficiently rare that it didn't seem worthwhile, at least until we get any feedback to the contrary.  But, if you think it's valuable, we could add it in.

Reply:
Agree, we can add that later as enhancement